attribute_rule_calculation/RotateFeatureByIntersectedLine.md

```diff
@@ -57,4 +57,130 @@ var segment = Clip(line, search)["paths"][0]
 
 // Get angle of line using the start and end vertex
 return Angle(segment[0], segment[-1])  
+```
+
+More complicated version that uses the number of lines to calculate the angle.  An example with using this rule is included in the [Example](./RotateFeatureByIntersectedLine.zip).  To demonstrate, add the RotationTest layer to an ArcGIS Pro project and create new lines and point features.
+```js
+// This calculation attribute rule intersects a line layer and based on the number of
+// intersected features, calculates the angle for the point
+
+// Set to true if the rotation setting is set to geographic in the layer properties
+var geographic_rotation = false;
+// Set the counter clockwise spin angle used for the symbol in the symbology options
+var symbol_flip_angle = 0
+// Return if a value is already set, to recalculate an angle, the field must be set to null
+if (IsEmpty($feature.symbolrotation) == false) {
+    return $feature.symbolrotation;
+}
+
+// Create a feature set to the line layer
+var lineClass = FeatureSetByName($datastore, "lines", ["objectid"], true);
+// Find the intersecting lines
+var lines = Intersects(lineClass, $feature);
+//If no lines intersect, return the original value
+if (lines == null) {
+    return $feature.symbolrotation;
+}
```

Comment:
This logic isn't working when the feature is not intersecting any lines.

Use:
if (Count(lines) == 0){
    return $feature.symbolrotation;
}

or move intersectcount above and use that.

Reply:
fixed